lib/api/model/model.dart

```diff
@@ -462,86 +463,142 @@ class Subscription extends ZulipStream {
+    StreamColor.barBackground:
+      LabColor.fromColor(const Color(0xfff9f9f9))
+        .interpolate(LabColor.fromColor(clamped20to75), 0.22)
+        .toColor(),
+  };
+
+  return ColorSwatch<StreamColor>(base, map);
+}
+
+/// A [ColorSwatch] with dark-mode colors related to a base stream color.
+///
+/// Use this in UI code for colors related to [Subscription.color],
+/// such as the background of an unread count badge.
+///
+/// For the light-mode colors, see [streamColorSwatchLight].
+ColorSwatch<StreamColor> streamColorSwatchDark(int base) {
+  final baseAsColor = Color(base);
+
+  final clamped20to75 = clampLchLightness(baseAsColor, 20, 75);
+
+  final map = {
+    StreamColor.base: baseAsColor,
+
+    // Follows `.unread-count` in Vlad's replit:
+    //   <https://replit.com/@VladKorobov/zulip-sidebar#script.js>
+    //   <https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/design.3A.20.23F117.20.22Inbox.22.20screen/near/1624484>
+    //
+    // TODO fix bug where our results differ from the replit's (see unit tests)
+    StreamColor.unreadCountBadgeBackground:
+      clampLchLightness(baseAsColor, 30, 70)
+        .withOpacity(0.3),
+
+    // Follows `.sidebar-row__icon` in Vlad's replit:
+    //   <https://replit.com/@VladKorobov/zulip-sidebar#script.js>
+    //
+    // TODO fix bug where our results differ from the replit's (see unit tests)
+    StreamColor.iconOnPlainBackground: clamped20to75,
+
+    // Follows the web app (as of zulip/zulip@db03369ac); see
+    // get_stream_privacy_icon_color in web/src/stream_color.ts.
+    //
+    // `.recepeient__icon` in Vlad's replit gives something different so we
+    // don't use that:
+    //   <https://replit.com/@VladKorobov/zulip-topic-feed-colors#script.js>
+    //   <https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/design.3A.20.23F117.20.22Inbox.22.20screen/near/1624484>
+    // But that's OK because Vlad said "I feel like current dark theme contrast
+    // is fine", and when he said that, this had been the web app's icon color
+    // for 6+ months (since zulip/zulip@023584e04):
+    //   https://chat.zulip.org/#narrow/stream/101-design/topic/UI.20redesign.3A.20recipient.20bar.20colors/near/1675786
+    //
+    // TODO fix bug where our results are unexpected (see unit tests)
+    StreamColor.iconOnBarBackground: clamped20to75,
+
+    // Follows `.recepient` in Vlad's replit:
+    //   <https://replit.com/@VladKorobov/zulip-topic-feed-colors#script.js>
+    //
+    // TODO I think [LabColor.interpolate] doesn't actually do LAB mixing;
+    //   it just calls up to the superclass method [ColorModel.interpolate]:
+    //     <https://pub.dev/documentation/flutter_color_models/latest/flutter_color_models/ColorModel/interpolate.html>
+    //   which does ordinary RGB mixing. Investigate and send a PR?
+    // TODO fix bug where our results differ from the replit's (see unit tests)
+    StreamColor.barBackground:
```

Comment:
This comment is all the same as in the light case, right?

Let's try to deduplicate comments that don't differ. It makes it easier to find the spots where there is new information in these comments (after reading the other version's comments), like for `iconOnBarBackground`.